src/aget_local.erl

```diff
@@ -109,6 +110,11 @@ get(AGetTable, Key, Default) ->
 current_ets(#?MODULE{cur_map = CurMap}) ->
     CurMap.
 
+-spec force_switch(#?MODULE{}) -> #?MODULE{}.
+force_switch(#?MODULE{cur_map = CurMap, prev_map = PrevMap} = State) ->
+    ets:delete_all_objects(PrevMap),
+    State#?MODULE{cur_map = PrevMap, prev_map = CurMap}.
```

Comment:
I think it is much better to just call `on_message` with `{aget_local, lifecycle}`. It will force switch maps and reset timer. Your `force_switch` is not resetting the timer, so it can trigger right after `force_switch` once again.
Sending `{aget_local, lifecycle}` would be better also if there are new changes in the logic.